WordPress/src/main/java/org/wordpress/android/ui/posts/EditPostViewModel.kt

```diff
@@ -0,0 +1,193 @@
+package org.wordpress.android.ui.posts
+
+import android.content.Context
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import kotlinx.coroutines.CoroutineDispatcher
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import org.wordpress.android.fluxc.Dispatcher
+import org.wordpress.android.fluxc.generated.PostActionBuilder
+import org.wordpress.android.fluxc.model.PostModel
+import org.wordpress.android.modules.UI_THREAD
+import org.wordpress.android.ui.posts.EditPostViewModel.UpdateFromEditor.PostFields
+import org.wordpress.android.ui.posts.EditPostViewModel.UpdateResult.Error
+import org.wordpress.android.ui.posts.EditPostViewModel.UpdateResult.Success
+import org.wordpress.android.util.AppLog
+import org.wordpress.android.util.DateTimeUtilsWrapper
+import org.wordpress.android.util.LocaleManagerWrapper
+import org.wordpress.android.viewmodel.Event
+import org.wordpress.android.viewmodel.ScopedViewModel
+import javax.inject.Inject
+import javax.inject.Named
+
+private const val CHANGE_SAVE_DELAY = 500L
+private const val MAX_UNSAVED_POSTS = 50
+
+class EditPostViewModel
+@Inject constructor(
+    @Named(UI_THREAD) private val mainDispatcher: CoroutineDispatcher,
+    private val dispatcher: Dispatcher,
+    private val aztecEditorWrapper: AztecEditorWrapper,
+    private val localeManagerWrapper: LocaleManagerWrapper,
+    private val dateTimeUtils: DateTimeUtilsWrapper
+) : ScopedViewModel(mainDispatcher) {
+    private var mDebounceCounter = 0
+    var mediaInsertedOnCreation: Boolean = false
+    var mediaMarkedUploadingOnStartIds: List<String> = listOf()
+    private val _onSavePostTriggered = MutableLiveData<Event<Unit>>()
+    val onSavePostTriggered: LiveData<Event<Unit>> = _onSavePostTriggered
+
+    fun savePost() {
+        launch {
```

Comment:
I didn't notice this during the review, but I think this doesn't work as intended. We don't cancel the previous job. I think we'll need to do something like the following
```
lastSavePostJob?.cancel()
       lastSavePostJob = launch {...}
```
Wdyt?

Reply:
This works the same way as it did before (I'm not saying it's correct) and performs the save operation multiple times. I agree though that cancelling the previous job is probably a more correct solution 👍

Reply:
I think it's a bug, we basically just postpone all saves by 500ms (we perform all of them)... the intended throttle doesn't work at all.

Reply:
I think you're right. I've implemented the job cancelling in https://github.com/wordpress-mobile/WordPress-Android/pull/10797/commits/e94a9266ec988a29b4af6f3cc03eb738af6b4a6c

Reply:
Awesome, thank you! ;) I'll test it one more time, before merging tomorrow.